protocol/binary/stream_writer.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package binary
+
+import (
+	"go.uber.org/thriftrw/protocol/stream"
+	"math"
+)
+
+// WriteBool encodes a boolean
+func (bw *Writer) WriteBool(b bool) error {
+	if b {
+		return bw.writeByte(1)
+	}
+	return bw.writeByte(0)
+}
+
+// WriteInt8 encodes an int8
+func (bw *Writer) WriteInt8(i int8) error {
+	return bw.writeByte(byte(i))
+}
+
+// WriteInt16 encodes an int16
+func (bw *Writer) WriteInt16(i int16) error {
+	return bw.writeInt16(i)
+}
+
+// WriteInt32 encodes an int32
+func (bw *Writer) WriteInt32(i int32) error {
+	return bw.writeInt32(i)
+}
+
+// WriteInt64 encodes an int64
+func (bw *Writer) WriteInt64(i int64) error {
+	return bw.writeInt64(i)
+}
+
+// WriteString encodes a string
+func (bw *Writer) WriteString(s string) error {
+	return bw.writeString(s)
+}
+
+// WriteDouble encodes a double
+func (bw *Writer) WriteDouble(d float64) error {
+	value := math.Float64bits(d)
+	return bw.writeInt64(int64(value))
```

Comment:
Could make this an unexported `writeDouble` that's also used by
```
case wire.TDouble:
```
since it's the same logic.